paasta_tools/long_running_service_tools.py

```diff
@@ -39,17 +40,39 @@
 # instantaneous CPU
 DEFAULT_CPU_AUTOSCALING_MOVING_AVERAGE_WINDOW = 60
 
+METRICS_PROVIDER_CPU = "cpu"
+METRICS_PROVIDER_UWSGI = "uwsgi"
+METRICS_PROVIDER_GUNICORN = "gunicorn"
+METRICS_PROVIDER_PISCINA = "piscina"
+METRICS_PROVIDER_ACTIVE_REQUESTS = "active-requests"
+METRICS_PROVIDER_PROMQL = "arbitrary_promql"
 
-class AutoscalingParamsDict(TypedDict, total=False):
-    metrics_provider: str
+
+def all_metrics_providers() -> List[str]:
+    return [
+        METRICS_PROVIDER_CPU,
+        METRICS_PROVIDER_UWSGI,
+        METRICS_PROVIDER_GUNICORN,
+        METRICS_PROVIDER_PISCINA,
+        METRICS_PROVIDER_ACTIVE_REQUESTS,
+        METRICS_PROVIDER_PROMQL,
+    ]
```

Comment:
question: any reason to not make this a global constant?

Reply:
I've been too conditioned by golangci-lint